flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNITCase.java

```diff
@@ -113,6 +118,13 @@ public void testPerJobMode() throws Exception {
 
 					assertThat(jobResult, is(notNullValue()));
 					assertThat(jobResult.getSerializedThrowable().isPresent(), is(false));
+
+					final FileSystem fs = FileSystem.get(getYarnConfiguration());
+					String suffix = ".flink/" + applicationId.toString() + "/" + flinkUberjar.getName();
+
+					Path uberJarHDFSPath = new Path(fs.getHomeDirectory(), suffix);
+					FileStatus fsStatus = fs.getFileStatus(uberJarHDFSPath);
+					Assert.assertEquals(5, fsStatus.getReplication());
```

Comment:
I think we should also check the replication of files uploaded by appMaster(flink resourcemanager).

Reply:
Not sure whether I understand your suggestions. This change mainly to add replica for those jar, libs, configs that are uploaded before start application master. As all of these files are go through the same function call, I think verify one of them is enough.

Reply:
Hi @HuangZhenQiu,
I think you made some changes of `Utils.createTaskExecutorContext()`. The replication of taskmanager-conf is set to `dfs.replication` and it need to be checked in the YarnITCase.

Reply:
Agree. As TM conf are new file for each TM, so I used default replication number. I added assert to check it in the new diff.